code/modules/power/generators/turbine.dm

```diff
@@ -338,9 +383,14 @@
 
 	switch(action)
 		if("toggle_power")
-			if(compressor?.turbine)
+			if(compressor?.turbine && compressor.last_overheat <= 0)
 				compressor.starter = !compressor.starter
 				. = TRUE
+				playsound(src, 'sound/mecha/powerup.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
+			if(compressor.last_overheat > 0)
+				compressor.starter = FALSE
+				to_chat(usr,"<span class='alert'>The turbine is overheating, please wait [compressor.last_overheat] seconds for cooldown procedures to complete.</span>")
+				playsound(src, 'sound/effects/electheart.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
```

Comment:
```suggestion
			var/time_til_done = compressor.time_until_overheat_done()
			if(time_til_done)
				compressor.starter = FALSE
				to_chat(usr,"<span class='alert'>The turbine is overheating, please wait [time_til_done] seconds for cooldown procedures to complete.</span>")
				playsound(src, 'sound/effects/electheart.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
			else if(compressor?.turbine)
				compressor.starter = !compressor.starter
				. = TRUE
				playsound(src, 'sound/mecha/powerup.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
```

Reply:
All comments addressed in latest commit: 
f555d19